applesingle.c

```diff
@@ -8,7 +8,7 @@ Build cmd for Mac OS 10.4:
 Build cmd for later releases (e.g. with openssl installed under /usr/local):
   cc -m32 -Wno-deprecated-declarations -O2 -framework Carbon -lcrypto applesingle.c -o applesingle
 Build cmd for Linux/x86:
-  cc -O2 -lcrypto applesingle.c -o applesingle
+  cc applesingle.c -O2 -lcrypto -o applesingle
```

Comment:
Can you help me understand the purpose of this change? It seems to have no effect on gcc...

Reply:
when building it the original way (on `Linux 5.4.0-58-generic #64-Ubuntu SMP x86_64 x86_64 x86_64 GNU/Linux`) i was getting the following linker error:

```
/usr/bin/ld: /tmp/cc95ZmtG.o: in function `main':
applesingle.c:(.text.startup+0xa5d): undefined reference to `MD5_Init'
/usr/bin/ld: applesingle.c:(.text.startup+0xb66): undefined reference to `MD5_Update'
/usr/bin/ld: applesingle.c:(.text.startup+0xbfa): undefined reference to `MD5_Final'
collect2: error: ld returned 1 exit status
```
after looking it up on https://stackoverflow.com/questions/1894013/how-to-use-openssl-with-gcc, it seemed like all the examples they used put the source file first, then the linker settings, and that seemed to fix it for me. I'm honestly still not sure why the different ordering fixed it for me

Reply:
It would help to state which compiler you're using. You may be able to obtain information that with "cc -v". Anyway, I've committed a change to rearrange the linker option. Thanks for letting me know about this build failure.

Reply:
here's my `cc -v` output if that helps:
```
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-17ubuntu1~20.04' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-HskZEa/gcc-9-9.3.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 
```